pkg/webhooks/scc/scc.go

```diff
@@ -0,0 +1,189 @@
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"security.openshift.io"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"securitycontextconstraints"},
+				Scope:       &scope,
+			},
+		},
+	}
+	defaultSCCs = []string{
+		"anyuid",
+		"hostaccess",
+		"hostmount-anyuid",
+		"hostnetwork",
+		"node-exporter",
+		"nonroot",
+		"privileged",
+		"restricted",
+		"pipelines-scc",
+	}
+)
+
+type SCCWebHook struct {
+	s runtime.Scheme
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *SCCWebHook {
+	scheme := runtime.NewScheme()
+	admissionv1.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	return &SCCWebHook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *SCCWebHook) Authorized(request admissionctl.Request) admissionctl.Response {
+	return s.authorized(request)
+}
+
+func (s *SCCWebHook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	scc, err := s.renderSCC(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a SCC from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	switch request.Operation {
+	case admissionv1.Delete:
+		if isDefaultSCC(scc) {
+			log.Info(fmt.Sprintf("Deleting operation detected on default SCC: %v", scc.Name))
+			ret = admissionctl.Denied(fmt.Sprintf("Deleting default SCCs %v is not allowed", defaultSCCs))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	case admissionv1.Update:
+		if isDefaultSCC(scc) {
+			log.Info(fmt.Sprintf("Updating operation detected on default SCC: %v", scc.Name))
+			ret = admissionctl.Denied(fmt.Sprintf("Modifying default SCCs %v is not allowed", defaultSCCs))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+	}
+
+	ret = admissionctl.Allowed("Request is allowed")
+	ret.UID = request.AdmissionRequest.UID
+	return ret
+}
+
+// renderSCC render the SCC object from the requests
+func (s *SCCWebHook) renderSCC(request admissionctl.Request) (*securityv1.SecurityContextConstraints, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	scc := &securityv1.SecurityContextConstraints{}
+
+	if len(request.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(request.OldObject, scc)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return scc, nil
+}
+
+// isDefaultSCC checks if the request is going to operate on the SCC in the
+// default list
+func isDefaultSCC(scc *securityv1.SecurityContextConstraints) bool {
+	for _, s := range defaultSCCs {
+		if scc.Name == s {
+			return true
+		}
+	}
+	return false
+}
+
+// GetURI implements Webhook interface
+func (s *SCCWebHook) GetURI() string {
+	return "/" + WebhookName
+}
+
+// Validate implements Webhook interface
+func (s *SCCWebHook) Validate(request admissionctl.Request) bool {
+	valid := true
+	valid = valid && (request.UserInfo.Username != "")
+	valid = valid && (request.Kind.Kind == "SecurityContextConstraints")
```

Comment:
@NautiluX I found the problem that why it does not work before. The resource scc has some different definition with other. The Kind here, generally the resources will use the singular, but it is plural here...

I have updated the webhook based on our discussion and tested on my staging cluster. Please help take a look.

Reply:
cool glad local testing is working now as well :)